### PR TITLE
build: use go install instead of depricated go get

### PR DESCRIPTION
**Pull request by iamniting**
ref: https://go.dev/doc/go-get-install-deprecation

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -210,20 +210,11 @@ jsonnet: ## Download jsonnet locally if necessary.
 GINKGO = $(shell pwd)/bin/ginkgo
 ginkgo: ## Download ginkgo and gomega locally if necessary.
 	$(call go-get-tool,$(GINKGO),github.com/onsi/ginkgo/v2/ginkgo)
-	$(call go-get-tool,$(GOMEGA),github.com/onsi/gomega/...)
```
Comment (leelavg) [inline Makefile:213]: - is removal of GOMEGA installation intentional?
- is it due to GINKGO also installing GOMEGA?

Comment (iamniting) [reply Makefile:213]: >is removal of GOMEGA installation intentional?
yes it is, because we don't require a binary of GOMEGA and we already have it in our go.mod

File: Makefile [hunk 2]
```diff
@@ -210,20 +210,11 @@ jsonnet: ## Download jsonnet locally if necessary.
 GINKGO = $(shell pwd)/bin/ginkgo
 ginkgo: ## Download ginkgo and gomega locally if necessary.
```
Comment (nbalacha) [inline Makefile:211]:
```suggestion
ginkgo: ## Download ginkgo locally if necessary.
```

**Comment by nbalacha** [comment]
Can you include this as a part of a PR to use go 1.18?

**Comment by iamniting** [comment]
@nbalacha As discussed over a meet updated the PR, I Will create a new PR for the 1.18 specifically

File: Makefile (hunk 2 above)
Comment (nbalacha) [reply Makefile:211]: Please update the comment.

File: Makefile [hunk 3]
```diff
@@ -197,7 +197,7 @@ controller-gen: ## Download controller-gen locally if necessary.
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.3)
```
Comment (nbalacha) [inline Makefile:200]: I'd recommend  putting this in a different commit and adding information about all the changes in the PR message.
commit 1: change go-get-tool to use go install
commit 2: update kustomize version
commit 3: remove ginkgo

If the go.mod changes are unrelated, please send a separate PR for them

File: go.mod [hunk 4]
```diff
@@ -48,6 +48,7 @@ require (
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
 	github.com/golang/protobuf v1.5.2 // indirect
 	github.com/google/gofuzz v1.1.0 // indirect
+	github.com/google/pprof v0.0.0-20210720184732-4bb14d4b1be1 // indirect
```
Comment (nbalacha) [inline go.mod:51]: If these are not related to the go install changes, please send a separate PR.

**Review by nbalacha — CHANGES_REQUESTED** [review]

File: go.mod (hunk 4 above)
Comment (iamniting) [reply go.mod]: removed them

File: Makefile (hunk 3 above)
Comment (iamniting) [reply Makefile:200]: done

**Review by nbalacha — APPROVED** [review]

**Review by sp98 — APPROVED** [review]

**Review by Yuggupta27 — APPROVED** [review]

**Comment by riya-singhal31** [comment]
Nitin can you please add the missing go.sum entry too in this PR. So that it can pass the e2e.
Ref:  https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_lvm-operator/187/pull-ci-red-hat-storage-lvm-operator-main-lvm-operator-bundle-e2e-aws/1542518880071585792

File: Makefile [hunk 5]
```diff
@@ -214,21 +214,12 @@ jsonnet: ## Download jsonnet locally if necessary.
 
 GINKGO = $(shell pwd)/bin/ginkgo
 ginkgo: ## Download ginkgo and gomega locally if necessary.
-	$(call go-get-tool,$(GINKGO),github.com/onsi/ginkgo/v2/ginkgo)
-	$(call go-get-tool,$(GOMEGA),github.com/onsi/gomega/...)
+	$(call go-get-tool,$(GINKGO),github.com/onsi/ginkgo/v2/ginkgo@v2)
```
Comment (nbalacha) [inline Makefile:217]: Should this use the latest version available? Currently it is https://github.com/onsi/ginkgo/releases/tag/v2.1.6

**Review by nbalacha — APPROVED** [review]
